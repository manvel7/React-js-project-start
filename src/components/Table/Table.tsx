import {TableHTMLAttributes} from "react"
import styled, {css, StyledProps} from "styled-components"
import {smallMixin} from "../../styles/typography"

export interface IProps extends TableHTMLAttributes<HTMLTableElement> {
  striped?: boolean
}

const stripedMixin = css<StyledProps<IProps>>`
  tbody {
    tr:nth-of-type(odd) {
      background-color: ${({theme}) => theme.palette.background.white};
    }
  }
`

const Table = styled.table<IProps>`
  border-collapse: separate;
  border-spacing: 0 0.8rem;
  color: ${({theme}) => theme.palette.text.dark};
  width: 100%;
  ${smallMixin<IProps>()}

  ${({striped}) => (striped ? stripedMixin : "")}
`

export default Table
