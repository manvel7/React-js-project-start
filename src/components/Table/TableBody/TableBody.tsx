import {HTMLAttributes} from "react"
import styled from "styled-components"

interface IProps extends HTMLAttributes<HTMLTableSectionElement> {}

const TableBody = styled.tbody<IProps>``

export default TableBody
