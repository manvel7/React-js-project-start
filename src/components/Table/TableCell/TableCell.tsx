import {HTMLAttributes} from "react"
import styled from "styled-components"

interface IProps extends HTMLAttributes<HTMLTableCellElement> {}

const TableCell = styled.td<IProps>`
  padding: 0.8rem;
`

export default TableCell
