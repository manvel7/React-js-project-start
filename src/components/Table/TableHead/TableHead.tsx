import {HTMLAttributes} from "react"
import styled from "styled-components"

interface IProps extends HTMLAttributes<HTMLTableRowElement> {}

const TableHead = styled.thead<IProps>`
  tr {
    background-color: transparent;
  }

  th {
    text-align: right;

    &:first-child {
      text-align: left;
    }
  }
`

export default TableHead
