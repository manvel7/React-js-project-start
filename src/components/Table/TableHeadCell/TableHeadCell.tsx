import {HTMLAttributes} from "react"
import styled from "styled-components"

interface IProps extends HTMLAttributes<HTMLTableHeaderCellElement> {}

const TableHeadCell = styled.th<IProps>`
  padding: 0 0.8rem;
`

export default TableHeadCell
