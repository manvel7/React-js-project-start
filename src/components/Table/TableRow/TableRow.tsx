import {HTMLAttributes} from "react"
import styled, {css, StyledProps} from "styled-components"

export interface IProps extends HTMLAttributes<HTMLTableRowElement> {
  disabled?: boolean
}

const disabledMixin = css<StyledProps<IProps>>`
  background-color: ${({theme}) =>
    theme.palette.background.disabled} !important;
  color: ${({theme}) => theme.palette.text.darkDimmed} !important;
`

const TableRow = styled.tr<IProps>`
  background-color: ${({theme}) => theme.palette.background.light};

  ${({disabled}) => (disabled ? disabledMixin : "")}

  td,th {
    text-align: right;

    &:first-child {
      text-align: left;
      border-radius: 0.4rem 0 0 0.4rem;
    }

    &:last-child {
      border-radius: 0 0.4rem 0.4rem 0;
    }
  }
`

export default TableRow
