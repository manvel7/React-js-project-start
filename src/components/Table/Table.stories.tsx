import React from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import Table, {IProps} from "./Table"
import TableBody from "./TableBody"
import TableCell from "./TableCell"
import TableHead from "./TableHead"
import TableHeadCell from "./TableHeadCell"
import TableRow from "./TableRow"

const meta: Meta<IProps> = {
  title: "Components/Table",
  component: Table,
  argTypes: {}
}

const TableStory: Story<IProps> = (args) => (
  <Table {...args}>
    <TableHead>
      <TableRow>
        <TableHeadCell>Species</TableHeadCell>
        <TableHeadCell>Log (m3/ha)</TableHeadCell>
        <TableHeadCell>Average height (m)</TableHeadCell>
        <TableHeadCell>Average diameter (cm)</TableHeadCell>
      </TableRow>
    </TableHead>
    <TableBody>
      <TableRow>
        <TableCell>Spruce</TableCell>
        <TableCell>290</TableCell>
        <TableCell>29</TableCell>
        <TableCell>234</TableCell>
      </TableRow>
      <TableRow>
        <TableCell>Dedicious</TableCell>
        <TableCell>329</TableCell>
        <TableCell>32</TableCell>
        <TableCell>329</TableCell>
      </TableRow>
      <TableRow disabled={true}>
        <TableCell>Spruce</TableCell>
        <TableCell>290</TableCell>
        <TableCell>29</TableCell>
        <TableCell>234</TableCell>
      </TableRow>
    </TableBody>
  </Table>
)

const Regular = TableStory.bind({})
Regular.args = {}

const Striped = TableStory.bind({})
Striped.args = {
  striped: true
}

export {Regular, Striped}

export default meta
