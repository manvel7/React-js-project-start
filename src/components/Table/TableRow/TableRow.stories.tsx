import React from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import Table from "../index"
import TableBody from "../TableBody"
import TableCell from "../TableCell/index"
import TableRow, {IProps} from "./TableRow"

const meta: Meta<IProps> = {
  title: "Components/Table/Row",
  component: TableRow,
  argTypes: {}
}

const TableRowStory: Story<IProps> = (args) => (
  <Table>
    <TableBody>
      <TableRow {...args}>
        <TableCell>Spruce</TableCell>
        <TableCell>290</TableCell>
        <TableCell>29</TableCell>
        <TableCell>234</TableCell>
      </TableRow>
    </TableBody>
  </Table>
)

const Regular = TableRowStory.bind({})
Regular.args = {}

const Disabled = TableRowStory.bind({})
Disabled.args = {
  disabled: true
}

export {Regular, Disabled}

export default meta
