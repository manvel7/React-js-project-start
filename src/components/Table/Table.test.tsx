import React from "react"
import {render} from "@testing-library/react"
import {UiProvider} from "../../styles"
import Table from "./Table"
import TableBody from "./TableBody"
import TableCell from "./TableCell"
import TableHead from "./TableHead"
import TableHeadCell from "./TableHeadCell"
import TableRow from "./TableRow"

describe("Table", () => {
  describe("Regular", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Table>
            <TableHead>
              <TableRow>
                <TableHeadCell>Species</TableHeadCell>
                <TableHeadCell>Log (m3/ha)</TableHeadCell>
                <TableHeadCell>Average height (m)</TableHeadCell>
                <TableHeadCell>Average diameter (cm)</TableHeadCell>
              </TableRow>
            </TableHead>
            <TableBody>
              <TableRow>
                <TableCell>Spruce</TableCell>
                <TableCell>290</TableCell>
                <TableCell>29</TableCell>
                <TableCell>234</TableCell>
              </TableRow>
              <TableRow>
                <TableCell>Dedicious</TableCell>
                <TableCell>329</TableCell>
                <TableCell>32</TableCell>
                <TableCell>329</TableCell>
              </TableRow>
              <TableRow disabled={true}>
                <TableCell>Spruce</TableCell>
                <TableCell>290</TableCell>
                <TableCell>29</TableCell>
                <TableCell>234</TableCell>
              </TableRow>
            </TableBody>
          </Table>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
  })

  describe("Striped", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Table striped={true}>
            <TableHead>
              <TableRow>
                <TableHeadCell>Species</TableHeadCell>
                <TableHeadCell>Log (m3/ha)</TableHeadCell>
                <TableHeadCell>Average height (m)</TableHeadCell>
                <TableHeadCell>Average diameter (cm)</TableHeadCell>
              </TableRow>
            </TableHead>
            <TableBody>
              <TableRow>
                <TableCell>Spruce</TableCell>
                <TableCell>290</TableCell>
                <TableCell>29</TableCell>
                <TableCell>234</TableCell>
              </TableRow>
              <TableRow>
                <TableCell>Dedicious</TableCell>
                <TableCell>329</TableCell>
                <TableCell>32</TableCell>
                <TableCell>329</TableCell>
              </TableRow>
              <TableRow disabled={true}>
                <TableCell>Spruce</TableCell>
                <TableCell>290</TableCell>
                <TableCell>29</TableCell>
                <TableCell>234</TableCell>
              </TableRow>
            </TableBody>
          </Table>
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })
  })
})
