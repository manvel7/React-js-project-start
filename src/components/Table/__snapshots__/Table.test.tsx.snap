// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Regular should match snapshot 1`] = `
.c0 {
  border-collapse: separate;
  border-spacing: 0 0.8rem;
  color: #232323;
  width: 100%;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

.c4 {
  padding: 0.8rem;
}

.c1 tr {
  background-color: transparent;
}

.c1 th {
  text-align: right;
}

.c1 th:first-child {
  text-align: left;
}

.c3 {
  padding: 0 0.8rem;
}

.c2 {
  background-color: #F7F7F7;
}

.c2 td,
.c2 th {
  text-align: right;
}

.c2 td:first-child,
.c2 th:first-child {
  text-align: left;
  border-radius: 0.4rem 0 0 0.4rem;
}

.c2 td:last-child,
.c2 th:last-child {
  border-radius: 0 0.4rem 0.4rem 0;
}

.c5 {
  background-color: #F7F7F7;
  background-color: #E6E6E6 !important;
  color: #808080 !important;
}

.c5 td,
.c5 th {
  text-align: right;
}

.c5 td:first-child,
.c5 th:first-child {
  text-align: left;
  border-radius: 0.4rem 0 0 0.4rem;
}

.c5 td:last-child,
.c5 th:last-child {
  border-radius: 0 0.4rem 0.4rem 0;
}

<table
  class="c0"
>
  <thead
    class="c1"
  >
    <tr
      class="c2"
    >
      <th
        class="c3"
      >
        Species
      </th>
      <th
        class="c3"
      >
        Log (m3/ha)
      </th>
      <th
        class="c3"
      >
        Average height (m)
      </th>
      <th
        class="c3"
      >
        Average diameter (cm)
      </th>
    </tr>
  </thead>
  <tbody
    class=""
  >
    <tr
      class="c2"
    >
      <td
        class="c4"
      >
        Spruce
      </td>
      <td
        class="c4"
      >
        290
      </td>
      <td
        class="c4"
      >
        29
      </td>
      <td
        class="c4"
      >
        234
      </td>
    </tr>
    <tr
      class="c2"
    >
      <td
        class="c4"
      >
        Dedicious
      </td>
      <td
        class="c4"
      >
        329
      </td>
      <td
        class="c4"
      >
        32
      </td>
      <td
        class="c4"
      >
        329
      </td>
    </tr>
    <tr
      class="c5"
      disabled=""
    >
      <td
        class="c4"
      >
        Spruce
      </td>
      <td
        class="c4"
      >
        290
      </td>
      <td
        class="c4"
      >
        29
      </td>
      <td
        class="c4"
      >
        234
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table Striped should match snapshot 1`] = `
.c0 {
  border-collapse: separate;
  border-spacing: 0 0.8rem;
  color: #232323;
  width: 100%;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

.c0 tbody tr:nth-of-type(odd) {
  background-color: #FEFEFE;
}

.c4 {
  padding: 0.8rem;
}

.c1 tr {
  background-color: transparent;
}

.c1 th {
  text-align: right;
}

.c1 th:first-child {
  text-align: left;
}

.c3 {
  padding: 0 0.8rem;
}

.c2 {
  background-color: #F7F7F7;
}

.c2 td,
.c2 th {
  text-align: right;
}

.c2 td:first-child,
.c2 th:first-child {
  text-align: left;
  border-radius: 0.4rem 0 0 0.4rem;
}

.c2 td:last-child,
.c2 th:last-child {
  border-radius: 0 0.4rem 0.4rem 0;
}

.c5 {
  background-color: #F7F7F7;
  background-color: #E6E6E6 !important;
  color: #808080 !important;
}

.c5 td,
.c5 th {
  text-align: right;
}

.c5 td:first-child,
.c5 th:first-child {
  text-align: left;
  border-radius: 0.4rem 0 0 0.4rem;
}

.c5 td:last-child,
.c5 th:last-child {
  border-radius: 0 0.4rem 0.4rem 0;
}

<table
  class="c0"
>
  <thead
    class="c1"
  >
    <tr
      class="c2"
    >
      <th
        class="c3"
      >
        Species
      </th>
      <th
        class="c3"
      >
        Log (m3/ha)
      </th>
      <th
        class="c3"
      >
        Average height (m)
      </th>
      <th
        class="c3"
      >
        Average diameter (cm)
      </th>
    </tr>
  </thead>
  <tbody
    class=""
  >
    <tr
      class="c2"
    >
      <td
        class="c4"
      >
        Spruce
      </td>
      <td
        class="c4"
      >
        290
      </td>
      <td
        class="c4"
      >
        29
      </td>
      <td
        class="c4"
      >
        234
      </td>
    </tr>
    <tr
      class="c2"
    >
      <td
        class="c4"
      >
        Dedicious
      </td>
      <td
        class="c4"
      >
        329
      </td>
      <td
        class="c4"
      >
        32
      </td>
      <td
        class="c4"
      >
        329
      </td>
    </tr>
    <tr
      class="c5"
      disabled=""
    >
      <td
        class="c4"
      >
        Spruce
      </td>
      <td
        class="c4"
      >
        290
      </td>
      <td
        class="c4"
      >
        29
      </td>
      <td
        class="c4"
      >
        234
      </td>
    </tr>
  </tbody>
</table>
`;
